src/GeneratorV2/Process/Overloaders/VectorOverloader.cs

```diff
@@ -0,0 +1,188 @@
+            }
+
+            bool foundVector = false;
+            NameTable nameTable = overload.NameTable.New();
+            Parameter[] parameters = overload.InputParameters;
+            List<Parameter> newParameters = parameters.ToList();
+            List<Parameter> pointerParameters = new List<Parameter>();
+            List<Parameter> vectorParameters = new List<Parameter>();
+            List<Parameter> countParameters = new List<Parameter>();
+            for (int i = 0; i < parameters.Length; i++)
+            {
+                Parameter parameter = parameters[i];
+                if (parameter.Type is CSPointer {BaseType: CSType csType} &&
+                    csType.TypeName == baseType)
+                {
+                    // Check if the length of the parameter is correct.
+                    int length = -1;
+                    if (parameter.Length is Constant constant)
+                    {
+                        length = constant.Value;
+                    }
+                    else if (parameter.Length is BinaryOperation binary)
+                    {
+                        // If we have a binary operation length we need to set the count parameter to 1.
+                        Expression other;
+                        if (binary.Left is Constant left)
+                        {
+                            length = left.Value;
+                            other = binary.Right;
+                        }
+                        else if (binary.Right is Constant right)
+                        {
+                            length = right.Value;
+                            other = binary.Left;
+                        }
+                        else
+                        {
+                            newOverloads = null;
+                            return false;
+                        }
+
+                        if (other is ParameterReference reference)
+                        {
+                            Parameter? countParameter = newParameters.Find(p => p.Name == reference.ParameterName);
+                            if (countParameter == null)
+                                throw new Exception($"{reference.ParameterName} is not a valid reference on {name}");
+                            countParameters.Add(countParameter);
+                        }
+                    }
+
+                    if (length < 2 || length > 4)
+                    {
+                        continue;
+                    }
+
+                    // Save the new parameter.
+                    nameTable.Rename(parameter, parameter.Name + "_ptr");
+                    string vectorType = "Vector" + length + (typeChar == 'f' ? "" : typeChar);
+                    newParameters[i] = parameter with
+                    {
+                        Type = new CSRef(csType.Constant ? CSRef.Type.In : CSRef.Type.Ref, new CSType(vectorType, csType.Constant)),
+                        Length = null
+                    };
```

Comment:
There are functions that don't take math types by `ref`, should we also overload them?
E.g.
```
void glUniform4f(GLint location,
                 GLfloat v0,
                 GLfloat v1,
                 GLfloat v2,
                 GLfloat v3);
```

Reply:
This is not a vector though. The vector variant is called glUniform4fv. Which is being overloaded IIRC

Reply:
I mean both a "vectors" depending on how you look at it. `glUniform4fv` sends the data by pointer, while `glUniform4f` send the vector by value on the stack.

You only *really* need one version of the overload for opentk math types though, but it's something to consider.